solr/core/src/java/org/apache/solr/cli/DeleteTool.java

```diff
@@ -102,13 +110,18 @@ public void runImpl(CommandLine cli) throws Exception {
   }
 
   protected void deleteCollection(CommandLine cli) throws Exception {
+    Http2SolrClient.Builder builder =
+        new Http2SolrClient.Builder()
+            .withIdleTimeout(30, TimeUnit.SECONDS)
+            .withConnectionTimeout(15, TimeUnit.SECONDS);
+    if (cli.hasOption("credentials")) {
```

Comment:
I think I wrote this 'if' a hundred times over. would it be simpler to have a `withOptionalBasicAuthCredentials` on the builder that would ignore null values instead of throwing some error? it would also allow for cleaner method chaining.

Reply:
I would be up for that...   Definitly need some sort of better way of dealing with this.   Maybe some more refactoring and move the method to `SolrCLI` instead of it being all over the place on the individual tools?

Reply:
Take a looksee at what I did?   THoughts?

Reply:
sorry missed this message. looks good!